src/main/scala/ScalaLirc.scala

```diff
@@ -78,3 +42,48 @@ class ScalaLirc(sourceFile: String = "/etc/lirc/lircd.conf") {
   }
 
 }
+
+object ScalaLirc {
+  def apply(sourceFile: String = "/etc/lirc/lircd.conf"): Option[ScalaLirc] = {
+    try {
+      val file = Source.fromFile(sourceFile)
+      val sourceLines = file.getLines.toList
+
+      val auxRemoteCodes: MutableMap[String, List[String]] = MutableMap()
+      val remoteQueue: MutableQueue[String] = MutableQueue()
+      var codes = false
+
+      sourceLines.foreach { line =>
+        val cleanLine = line.replace("\t", " ").trim
+        cleanLine match {
+          case "begin remote" => remoteQueue.clear
+          case nameLine if cleanLine.indexOf("name") == 0 =>
+            // enqueue the remote name
+            remoteQueue.enqueue(nameLine.split(" ").last)
+          case "begin codes" => codes = true
+          case "end codes" => codes = false
+          case "end remote" =>
+            // move everything to the map
+            auxRemoteCodes += (remoteQueue.dequeue -> remoteQueue.toList)
+          case codeLine if codes == true =>
+            // enqueue the remote code
+            remoteQueue.enqueue(codeLine.split(" ").head)
+          case _ =>
+        }
+      }
```

Comment:
All the work should be done before trying to instantiate any `ScalaLirc` object :+1: Once we know that the file can actually be parsed, and we extracted all the codes from it, saving them in `auxRemoteCodes`, then we know that a `ScalaLirc` object can exist with all of its functionalities.